receiver/prometheusreceiver/internal/metadata.go

```diff
@@ -45,7 +49,7 @@ func (t *mService) Get(job, instance string) (MetadataCache, error) {
 
 	// from the same targetGroup, instance is not going to be duplicated
 	for _, target := range targetGroup {
-		if target.DiscoveredLabels().Get(model.AddressLabel) == instance {
+		if target.Labels().Get(model.InstanceLabel) == instance {
```

Comment:
2 changes:
- Replaced `DiscoveredLabels` replaced with `Labels` which contain effective label values after relabeling
- Matching against `instance` label instead of `__address__`: they are equal by default, but may differ after relabeling or when explicitly set to different values

Reply:
Note that I haven't switched the `SharedLabels()` implementation from `DiscoveredLabels` to `Labels` as previously suggested in https://github.com/open-telemetry/opentelemetry-collector/pull/710/files#r409196772

The reason is that Prometheus' `Labels()` method skips all reserved labels for some reason, see https://github.com/prometheus/prometheus/blob/cd12f0873c3eb2031f7ba9b2e169449aa1012e3f/scrape/target.go#L175 so it would potentially break some functionality.